sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -944,24 +959,12 @@ public void validate(PCollection<T> input) {
         }
       }
 
-      if (input.isBounded() == PCollection.IsBounded.UNBOUNDED || getTableRefFunction() != null) {
+      if (input.isBounded() == PCollection.IsBounded.UNBOUNDED) {
         // We will use BigQuery's streaming write API -- validate supported dispositions.
```

Comment:
Because it's no longer accurate - we use tableFunction in all cases as of this PR, so this assertion will now break existing use cases. Furthermore it was never a good check - we've had users complain about this check, and in fact one of them has an active PR to remove it.